ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/handler/AbortedTxnCleaner.java

```diff
@@ -79,8 +79,8 @@ public List<Runnable> getTasks() throws MetaException {
     List<CompactionInfo> readyToCleanAborts = txnHandler.findReadyToCleanAborts(abortedTimeThreshold, abortedThreshold);
 
     if (!readyToCleanAborts.isEmpty()) {
-      return readyToCleanAborts.stream().map(ci -> ThrowingRunnable.unchecked(() ->
-                      clean(ci, ci.txnId > 0 ? ci.txnId : Long.MAX_VALUE, metricsEnabled)))
+      return readyToCleanAborts.stream().map(info -> ThrowingRunnable.unchecked(() ->
+                      clean(info, info.txnId, metricsEnabled)))
```

Comment:
i think we should use `CQ_NEXT_TXN_ID` = (minOpenWriteTxn - 1) instead of CQ_TXN_ID, to be consistent (hwm for committed write txn).
and then minOpenWriteTxn = ci.nextTxnId + 1

Reply:
As discussed, introduced a separate field called `minOpenWriteTxnId` in compaction info. Done.